### PR TITLE
Reinstate IDE specific troubleshooting info

### PR DESCRIPTION
**Pull request by micgibso**
Signed-off-by: micgibso <MICGIBSO@uk.ibm.com>

PR for issue https://github.com/eclipse/codewind/issues/2633

This PR reinstates the IDE specific help for Eclipse, VS Code, and Che. I've reinstated it, edited it, and run the linkchecker.bat against it correcting any troubleshooting link errors.

@tetchel @eharris369 Please review, thx.

### CODE REVIEW

File: docs/_documentations/troubleshooting.md [hunk 1]
```diff
@@ -849,4 +857,149 @@ Command options are:
...
+
+If the debugger is connected but stopped on a ClassNotFoundException, click on the run button to get past the exception. You might need to click run several times as the exception occurs more than once. To avoid stopping on the exception in the future, open the Eclipse preferences and navigate to **Java** > **Debug**. Uncheck **Suspend execution on uncaught exceptions** and click **Apply and Close**.
+
+If the application is not waiting for the debugger to connect, try restarting the application again. If this does not work, use Codewind to disable the application and then re-enable it.
+
+## Application does not rebuild after making a change
+To start a build manually, right click on the application in the **Codewind Explorer** view, and selecting **Build**. 
+
+## Correct project list is not being shown
+Try refreshing by right-clicking on the **Local Projects** item in the **Codewind Explorer** view and selecting **Refresh**.
```
Comment (eharris369) [inline docs/_documentations/troubleshooting.md:909]: Change to: 
Try refreshing by right-clicking on the connection in the **Codewind Explorer** view and selecting **Refresh**.

Comment (sishida) [reply docs/_documentations/troubleshooting.md:909]: Grammar/style recommendations:
- Change to "Refresh by right-clicking the..."
- Remove "Try" to make the action sound more definitive.
- Remove "on" as stated in the style guidelines (https://github.com/eclipse/codewind-docs/wiki/Writing-style-guidelines).

File: docs/_documentations/troubleshooting.md [hunk 2]
```diff
@@ -849,4 +857,149 @@ Command options are:
...
+## Correct project list is not being shown
+Try refreshing by right-clicking on the **Local Projects** item in the **Codewind Explorer** view and selecting **Refresh**. 
+
+## Application is not showing the correct status
+Try refreshing the application by right-clicking on it in the **Codewind Explorer** view and selecting **Refresh**. 
+
+***
+# VS Code troubleshooting
+
+When troubleshooting, check the project in Codewind to make sure the status is what you expect. For Codewind specific problem solving tips when using VS Code, see the following information.
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:917]: > When troubleshooting, check the project in Codewind to make sure the status is what you expect.

I'm not sure what this line means, really.

File: docs/_documentations/troubleshooting.md [hunk 3]
```diff
@@ -849,4 +857,149 @@ Command options are:
...
+***
+# Che troubleshooting
+
+When troubleshooting, check the project in Codewind to make sure the status is what you expect. For Codewind specific problem solving tips when using Che, see the following information.
+
+* [Check the Che logs](#check-the-che-logs)
+* [Solving common Che problems](#solving-common-che-problems)
+
+## Check the Che logs
+The logs are found in your Eclipse workspace under *.metadata/.log*.
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:971]: This is not accurate, those are the Eclipse IDE logs. You can just remove this section.

File: docs/_documentations/troubleshooting.md [hunk 4]
```diff
@@ -849,4 +857,149 @@ Command options are:
...
+
+## Solving common Che problems
+The following list describes common problems that might affect Codewind.
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:974]: All of these appear to apply to the Eclipse IDE, not Che. I don't think we have/had a Che troubleshooting section.

File: docs/_documentations/troubleshooting.md (hunk 1 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:909]: Corrected.

File: docs/_documentations/troubleshooting.md (hunk 3 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:971]: Removed.

File: docs/_documentations/troubleshooting.md (hunk 4 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:974]: Found a very old file called mdt-che-troubleshooting.md so put it's contents in here, but as you say, it's an Eclipse IDE copy. Removed.

File: docs/_documentations/troubleshooting.md (hunk 2 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:916]: Deleted line.

**Comment by micgibso** [comment]
@tetchel @eharris369 All comments incorporated.

**Comment by tetchel** [comment]
Can you move the `Executable file not found on PATH in VS Code` section into the VS Code troubleshooting section?

**Comment by micgibso** [comment]
@tetchel Hi Tim, moved issue into VS Code section, thanks.

**Comment by micgibso** [comment]
Resolved merge conflicts.

**Comment by micgibso** [comment]
@tetchel Hi Tim, moved issue into VS Code section, edited, viewed locally, tweaked ordering and resolved display issues. thanks.

File: docs/_documentations/troubleshooting.md [hunk 5]
```diff
@@ -799,7 +805,119 @@ To assist with problem determination, raise the default Codewind server log leve
...
+- [Codewind output stream](#codewind-output-stream)
+- [Finding the extension logs](#finding-the-extension-logs)
+- [Executable file not found on PATH in VS Code](#executable-file-not-found-on-path-in-vs-code)
+- [Build succeeded, but project is stuck in the Stopped state](#build-succeeded-but-project-is-stuck-in-the-stopped-state)
+- [No ESLint warnings or errors](#no-eslint-warnings-or-errors)
+- [Debug](#debug)
+
+## Codewind output stream
+
+The Codewind output stream is available in the VS Code editor. It logs `cwctl` commands together with their ouput.
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:904]: Typo ouput

File: docs/_documentations/troubleshooting.md [hunk 6]
```diff
@@ -827,28 +945,21 @@ If you can run `docker` from the command line, but Codewind still fails to find
...
 4. Now, the new VS Code instance shares the terminal's `PATH`. Start Codewind again.
 
-<!--
-Action/Topic: Collecting log files and environment data
-Issue type: info
-Issue link: https://github.com/eclipse/codewind/issues/2766
-Info added in 0.12.0.
--->
-# Collecting log files and environment data
+## Build succeeded but project is stuck in the Stopped state
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:948]: This is more general troubleshooting than vscode-specific

File: docs/_documentations/troubleshooting.md [hunk 7]
```diff
@@ -827,28 +945,21 @@ If you can run `docker` from the command line, but Codewind still fails to find
...
-Command options are:
-* `--conid remote` - Triggers diagnostics collection for the remote codewind instance (_must_ have currently configured Kubectl connection, default:"local")
-* `--eclipseWorkspaceDir/-e <value>` - The location of your Eclipse workspace directory if using the Eclipse IDE, default:"")
-* `--intellijLogsDir/-i <value>` - The location of your IntelliJ logs directory if using the IntelliJ IDE (default: "")
-* `--quiet/-q` - Turn off console messages
-* `--projects/-p` - Collect project containers information
-* `--nozip/-n` - Does not create collection zip and leaves individual collected files in place
-* `--clean` - Removes the `diagnostics` directory and all its contents from the Codewind home directory
+### Liberty project remains in the Starting - Debug state
+Liberty servers started in **Debug** mode do not start until the debugger attaches. Run the `attach debugger` command] and the server starts. You can check the application logs to see if the server is starting.
```
Comment (tetchel) [inline docs/_documentations/troubleshooting.md:959]: This is also general troubleshooting

File: docs/_documentations/troubleshooting.md (hunk 5 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:904]: Corrected

File: docs/_documentations/troubleshooting.md (hunk 6 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:948]: This is a very old Ts issue. Searched the Codewind issues and found only two that were both closed, so have removed this. The issue is covered by the `## Projects stuck in starting or stopped state` issue in the main troubleshooting section.

File: docs/_documentations/troubleshooting.md (hunk 7 above)
Comment (micgibso) [reply docs/_documentations/troubleshooting.md:959]: This is a very old Ts item for which I cannot find a corresponding issue. The remedial action -attach debugger- is covered in several places throughout the Ts, so removing.

**Comment by micgibso** [comment]
@tetchel All comments resolved, please review, thx!

**Pull request merged**